operator/docs/enhancements/zone_aware_replication.md

```diff
@@ -0,0 +1,270 @@
+---
+title: Zone Awareness Replication Support
+authors:
+  - "@shwetaap"
+  - "@btaani"
+reviewers:
+  - "@xperimental"
+  - "@periklis"
+draft: false
+menu:
+  docs:
+    parent: "enhancements"
+weight: 100
+toc: true
+---
+
+# Table of Contents
+1. [Summary](#summary)
+2. [Motivation](#motivation)
+3. [Proposal](#proposal)
+    1. [API Extensions](#api-extensions)
+    2. [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+4. [Design Details](#design-details)
+5. [Impact on LokiStack Sizing Configurations](#sizing)
+
+## Summary
+
+By default, data is transparently replicated across the whole pool of service instances, regardless of whether these instances are all running within the same availability zone (or data center, or rack) or in different ones. Storing multiple replicas for a given data within the same availability zone poses a risk for data loss if there is an outage affecting various nodes within a zone or a full zone outage. There is support for Zone Aware Replication in Loki. When enabled, replicas for the given data are guaranteed to span across different availability zones.
+A zone represents a logical failure domain. It is common for Kubernetes clusters to span multiple zones for increased availability. While the exact definition of a zone is left to infrastructure implementations, common properties of a zone include very low network latency within a zone, no-cost network traffic within a zone, and failure independence from other zones.
+The following sections describe a set of APIs in form of Custom Resource Definitions (CRD) that enable users of `LokiStack` resources to support:
+- Enable Zone-Aware Replication configuration in LokiStack CR so the components are deployed in different zones.
+- Configure `loki-config.yaml` to support zone-aware data replication.
+
+## Motivation
+
+Zone-aware replication is the replication of data across failure domains. Avoiding data loss during a domain outage is the motivation to introduce zone-aware component deployment and data replication. 
+
+### Goals
+
+* The user can enable zone aware replication for the read and the write path
```

Comment:
@cstyan Thanks for shedding more light on this. FWIW the Loki Operator deploys the Loki components in all-or-nothing fashion. Means it simply patches all Deployments/StatefulSets without considering if the belong to the write or the read path. So far we kept off our hands from doing this because it simplifies the design of the operator being a simple manifest create-or-update loop (plus some PKI certs generation).

@shwetaap The key point raised by @cstyan here needs explicit mention on the design proposal imho. For the sake of sanity and later documentation efforts we need to highlight that:
1. Either the operator updates the paths in two step, where waiting the max-query-lookback period (currently `30s` IIRC). 
2. Or the users should expect for the same period of time missing queries in the data.

Reply:
Made a note of this here  - https://github.com/grafana/loki/blob/48ae6628e1508ceac3bfd897cc1ec52c66d651b6/operator/docs/enhancements/zone_aware_replication.md#1-read-and-write-path